### DISCUSSION TITLE
[Feature] Delete all tuples related to "anytype:id123"

### DISCUSSION THREAD

**Discussion by xXAvoraXx** [openfga/community#225] [Ideas]
Deleting a tuple can sometimes get very complicated. For example, if I want to delete all relationship tuples with "id123" of type "anytype:id123", I need to request all types that have a relationship with "anytype" with list endpoint and sometimes we may miss some tuples. In short, I think a garbage collection feature is needed, for example delete all tuples related to "anytype:id123".

**Comment by jon-whit** [comment]
@xXAvoraXx are you specifically asking for the ability to filter tuples in a more granular way (e.g. list all tuples with `object = anytype:id123`)? 

We're aware that the Read API has some non-idealisms in it's current design and we have discussed an improved Read API that will provide more granular filtering capabilities.

**Reply by xXAvoraXx** [reply]
For example, I would like to be able to delete all the tuples that an object that is no longer used in the system is associated with in one go.

Suppose we have tuples like the one below.

```
{
 user: user:john
 relation: can_access
 object: toys:123,
},
{
 user: user:anne
 relation: can_access
 object: toys:123,
},
{
 user: user:alice
 relation: can_access
 object: toys:123,
},
{
user: toys:123
relation: member
object: category:1
}
```

If I delete the toys:123 object from the database of my application, there is no need to keep the tuples of toys:123 in the openfga database, so the garbage of toys:123 should be collected and deleted.

If I make a call to the openfga api like below, all the tuples above should be deleted.

deleteTuplesByTypeId("toys:123");

**Comment by aaguiarz** [comment]
@xXAvoraXx this idea is now captured in our https://github.com/openfga/roadmap/issues/34

**Reply by xXAvoraXx** [reply]
That's good to hear.

**Reply by xXAvoraXx** [reply]
@aaguiarz Hi, is there any updates?
